lib/store/task.ts

```diff
@@ -67,4 +68,53 @@ export const useTaskStore = create<TaskStore>((set) => ({
       }),
     }));
   },
+
+  /**
+   * Reorder a task to the position of another task. This function will update
+   * the position of the task in the Dexie database and then update the store
+   * with the moved task.
+   *
+   * @param {string} activeId The id of the task being dragged.
+   * @param {string} overId The id of the task being hovered over.
+   * @returns {void}
+   */
+  reorderTask: async (activeId, overId) => {
+    const tasks = await taskService.getAllTasks();
+
+    // find index of task being dragged and task being hovered over
+    const activeIndex = tasks.findIndex((task) => task.id === activeId);
+    const overIndex = tasks.findIndex((task) => task.id === overId);
+
+    if (activeIndex === -1 || overIndex === -1 || activeIndex === overIndex) {
+      return; // no op drag event
+    }
+
+    // reorder the array
+    const [movedTask] = tasks.splice(activeIndex, 1);
+    tasks.splice(overIndex, 0, movedTask);
+
+    // calculate the new position for moved task
+    const prevTask = tasks[overIndex - 1];
+    const nextTask = tasks[overIndex + 1];
+
+    let newPosition;
+
+    if (prevTask && nextTask) {
+      newPosition = (prevTask.position + nextTask.position) / 2;
+    } else if (prevTask) {
+      newPosition = prevTask.position + 1; // Place at the end
+    } else if (nextTask) {
+      newPosition = nextTask.position / 2; // place at the start
+    } else {
+      newPosition = 1; // fallback to place at start
+    }
```

Comment:
I think this logic is kind of messy - reassigning a variable probably isn't the cleanest way to get this done 🤔